### DISCUSSION TITLE
Using v2.0 with class based components?

### DISCUSSION THREAD

**Discussion by alexfoxy** [software-mansion/react-native-reanimated#1800] [Q&A]
Is there any documentation on how to use reanimated 2.0 with class based components? For example what is the equivalent of `useSharedValue`, `useAnimatedStyle` & `useAnimatedScrollHandler` ?

I have had a look around on the issues and dug into the code a little but I couldn't make much progress. Is this possible?

Thanks!
Alex

**Comment by mrousavy** [comment]
It's not possible without a big refactor of the code. You'll have to use hooks for this.

**Reply by zillBoy** [reply]
Ooh god, that's going to take a long time. I've a really complex class component 2500 lines of code 😵‍💫😬
